openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -343,12 +343,10 @@ def _track_user_registration(user, profile, params, third_party_provider):
                 'email': user.email,
                 'label': params.get('course_id'),
                 'provider': third_party_provider.name if third_party_provider else None,
-                'optional_fields': bool(
-                    profile.goals
-                    or profile.level_of_education_display
-                    or profile.gender_display
-                    or profile.year_of_birth
-                )
+                'gender_field': bool(profile.gender_display),
```

Comment:
I guess we can rename these to more meaningful names, `gender_field` does not explain what we are capturing. e.g. `is_gender_selected`